js/components/Graph/Graph.js

```diff
@@ -0,0 +1,371 @@
+    for (let i = 1; i <= 8; i++) {
+      allButtons.push(
+        <TouchableOpacity
+          key={i}
+          activeOpacity={0.6}
+          style={[
+            styles.graphButton,
+            i === 1
+              ? this.state.joy
+                ? { backgroundColor: "#78D3FD" }
+                : { backgroundColor: "#F3F0EC" }
+              : i === 2
+              ? this.state.sadness
+                ? { backgroundColor: "#D0C4ED" }
+                : { backgroundColor: "#F3F0EC" }
+              : i === 3
+              ? this.state.comfort
+                ? { backgroundColor: "#FEC1C2" }
+                : { backgroundColor: "#F3F0EC" }
+              : i === 4
+              ? this.state.fear
+                ? { backgroundColor: "#FFEDC3" }
+                : { backgroundColor: "#F3F0EC" }
+              : i === 5
+              ? this.state.esteem
+                ? { backgroundColor: "#B2EDD2" }
+                : { backgroundColor: "#F3F0EC" }
+              : i === 6
+              ? this.state.shame
+                ? { backgroundColor: "#FFCAF8" }
+                : { backgroundColor: "#F3F0EC" }
+              : i === 7
+              ? this.state.anger
+                ? { backgroundColor: "#C9E0F1" }
+                : { backgroundColor: "#F3F0EC" }
+              : i === 8 && this.state.calmness
+              ? { backgroundColor: "#C6CACC" }
+              : { backgroundColor: "#F3F0EC" }
+          ]}
+          onPress={() => {
+            i === 1
+              ? this.setState({ joy: !this.state.joy })
+              : i === 2
+              ? this.setState({ sadness: !this.state.sadness })
+              : i === 3
+              ? this.setState({ comfort: !this.state.comfort })
+              : i === 4
+              ? this.setState({ fear: !this.state.fear })
+              : i === 5
+              ? this.setState({ esteem: !this.state.esteem })
+              : i === 6
+              ? this.setState({ shame: !this.state.shame })
+              : i === 7
+              ? this.setState({ anger: !this.state.anger })
+              : i === 8 && this.setState({ calmness: !this.state.calmness });
+          }}
+        >
+          {i === 1 ? (
+            <Text style={styles.graphButtonText}>Joy</Text>
+          ) : i === 2 ? (
+            <Text style={styles.graphButtonText}>Sadness</Text>
+          ) : i === 3 ? (
+            <Text style={styles.graphButtonText}>Comfort</Text>
+          ) : i === 4 ? (
+            <Text style={styles.graphButtonText}>Fear</Text>
+          ) : i === 5 ? (
+            <Text style={styles.graphButtonText}>Esteem</Text>
+          ) : i === 6 ? (
+            <Text style={styles.graphButtonText}>Shame</Text>
+          ) : i === 7 ? (
+            <Text style={styles.graphButtonText}>Anger</Text>
+          ) : (
+            i === 8 && <Text style={styles.graphButtonText}>Calmness</Text>
+          )}
+        </TouchableOpacity>
+      );
+    }
+    return allButtons;
```

Comment:
maybe there is a better way to do this?

Reply:
extra ternary practice and it's a function that makes all my buttons instead of making jsx for each.